packages/e2e-test-utils/src/templates.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Internal dependencies
+ */
+import { rest, batch } from './rest-api';
+
+const PATH_MAPPING = {
+	wp_template: '/wp/v2/templates',
+	wp_template_part: '/wp/v2/template-parts',
+};
+
+/**
+ * Delete all the templates of given type.
+ *
+ * @param {('wp_template'|'wp_template_part')} type - Template type to delete.
+ */
+export async function deleteAllTemplates( type ) {
+	const path = PATH_MAPPING[ type ];
+
+	if ( ! path ) {
+		return;
+	}
+
+	const templates = await rest( { path } );
+
+	await batch(
+		templates
+			.filter( ( template ) => !! template.wp_id )
+			.map( ( template ) => ( {
+				method: 'DELETE',
+				path: `/wp/v2/posts/${ template.wp_id }?force=true`,
```

Comment:
This is not working properly in my tests

Reply:
Interesting. I'll have a look tomorrow or Monday.

Are you testing this util separately, or e2e tests are failing?

Reply:
I was debugging the template-part test in my PR and noticed that the template parts were not removed between runs.

Reply:
One limitation of this util (and in some parts of templates endpoint) is that it can only delete templates for the current theme.

But I will see if there are any other issues. If util isn't reliable, we can always revert to my original PR - #38486.